### DISCUSSION TITLE
Mensuration signs with mixed notation

### DISCUSSION THREAD

**Discussion by ahankinson** [music-encoding/music-encoding#864] [General]
This continues a conversation that sprung up in PR #848 . I'm going to selectively quote a few posts here so we can continue the discussion, rather than fill up the PR.

**Comment by ahankinson** [comment]
@benory said:

> @ahankinson, thanks for your reply! I've now taken a look at the test suite examples, and two things come to mind. First, while the mensuration signs look very reasonable graphically, I want to be clear that my project uses mensuration signs as purely graphical symbols that the computer does not need to know the meaning of. The need for the signs is primarily to indicate tempo and metric information. Second—following my first point—I am not showing the music in mensural notation; I use mensural symbols together with modern notation. I imagine that it is possible to draw the symbols from this suite, once implemented, without using mensural notation?

**Reply by ahankinson** [reply]
Verovio engraves the MEI assuming that everything has "meaning". Verovio isn't a graphics editor, and MEI isn't a graphics encoding system. So encoding a mensuration sign will make Verovio think you're doing mensural notation. 

That said, you can embed SVG graphics in MEI. Verovio would likely just ignore it, but you can also program your display software to extract any embedded SVG from an MEI file and display it as you like. This would probably take a bit of custom JavaScript on your part.

Verovio does have a 'library' of SVG musical symbols, of sorts, that you could use. The [SMuFL range for Mensural Prolations](https://www.smufl.org/version/latest/range/medievalAndRenaissanceProlations/) gives the range and specific code points (e.g., "E910") which you can then find in the Verovio 'fonts' directory: https://github.com/rism-digital/verovio/tree/develop/data/Leipzig (look for "E910-mensuralProlation1.xml").

**Comment by ahankinson** [comment]
@craigsapp said:

> That will not work.  In the JRP I use the graphical mensuration symbols to categorize the music for analysis.  In CMN there is no longer any intrinsic rhythm information (other than to encode how to reverse the process back to mensural music), but it is a categorical symbol and has a meaning in that sense.   Here is an example of such use:

https://josquin.stanford.edu/analysis/rhythm

<img width="1392" alt="Screen Shot 2021-10-15 at 9 58 10 AM" src="https://user-images.githubusercontent.com/3487289/137525061-b33fae90-fb29-4b72-94a2-05ed5a5fef5d.png">

This is a list of histograms of rhythm patterns (at the breve level) under various mensuration signs which in turn are used to analyze stylistic patterns in various composers.   Describing mensural music in terms of modern time signatures is of not much use for such analyses since there will be one modern time signature that could be more than one possibility of a mensuration sign.  Showing the mensuration sign also implies that no rhythmic reduction has been applied to the notes, which has be a common thing to do in 20th century editions of mensural music (dividing the durations by 1/2 or 1/4 to make the half note or quarter note the beat rather than the original use of the whole-note for the beat).

When a person looks at the music in modern notation, also seeing modern time signatures conveys no particularly useful information that can be worked back to the mensural notation.  In that sense, @benory's problem is not application specific.  For example the New Josquin Edition (published from 1980's to current):
     https://www.kvnm.nl/en/Publisher/The-New-Josquin-Edition
uses the same style: resolving the relative durations of mensural music into the absolute durations of CMN, but keeping the mensural signs:
     https://www.kvnm.nl/content/finder-upload/files/NJE%20informatie.pdf

<img width="746" alt="Screen Shot 2021-10-15 at 10 05 14 AM" src="https://user-images.githubusercontent.com/3487289/137525931-63bf9cbf-94ae-4be2-8732-f3354215c9de.png">


--------------------

Related to the display of mensural signs in CMN here is an example of a `3`  (mensuration?  as it seems to have a different meaning here than in the music from 1450-1550 where `3` cannot be the starting mensuration sign) from the Fitzwilliam's Virginal Book (music composed around 1600 give or take a few decades):

https://imslp.org/wiki/Fitzwilliam_Virginal_Book_(Various)
https://www.kvnm.nl/content/finder-upload/files/NJE%20informatie.pdf

<img width="635" alt="Screen Shot 2021-10-15 at 10 26 27 AM" src="https://user-images.githubusercontent.com/3487289/137528607-2f688fbf-16e0-49bc-9836-e2e18aa2ac01.png">

<img width="655" alt="Screen Shot 2021-10-15 at 10 32 18 AM" src="https://user-images.githubusercontent.com/3487289/137529168-49be252c-74e2-49e3-a59d-bfc879234efd.png">

https://en.wikipedia.org/wiki/Fitzwilliam_Virginal_Book

This is from an 1899 edition where they give the original mensuration sign along with a modern interpretation in brackets after it. (but why there is a 9/4 and 3/4 measure in 3/2, I do not know...).

Other example showing the original C-dot mensuration:

<img width="647" alt="Screen Shot 2021-10-15 at 10 36 17 AM" src="https://user-images.githubusercontent.com/3487289/137529575-5d7630f8-550b-4a48-b6e2-5a62cfe60733.png">

**Reply by ahankinson** [reply]
>  ... In the JRP I use the graphical mensuration symbols to categorize the music for analysis ... 

I presume you could just as easily embed some other metadata in the MEI that will help also categorize it for analysis? Using, for example, items in a [`<classification>`](https://music-encoding.org/guidelines/dev/elements/classification.html) section? In the example you show it doesn't even use the graphical symbol, just a textual representation of what it is. 

Music notation is fundamentally a picture that conveys some sort of meaning. You can draw anything you like on a picture, and a person will then interpret the meaning. What I hear you asking for is a semantically-encoded image where the encoding supplies some sort of meaning.

An encoding system that can encapsulate all possible semantic meanings that you can possibly draw in an image will be very difficult to manage. I think we have to assume a border between what we try to encode, structurally, and what we simply need to display, visually. Mixing mensural and modern notations where the mensural notations have no "meaning", and the modern do, would be very challenging to implement. 

This is why I keep suggesting embedded SVG as a possible solution. Using SVG conveys absolutely no semantic meaning, so you can put anything you like in it, and the encoding doesn't have to try and interpret an increasing number of (possibly conflicting) semantic meanings.

Put another way, if the only purpose of "encoding" the mensuration symbol is so that Verovio can just select the appropriate SVG symbol and embed it in the score without affecting any other component of the rendering (timing / alignment / etc.) then why not cut out the middle and just embed the SVG directly? Or embed the SMuFL code point with a symbol?

**Comment by craigsapp** [comment]
Also relevant is this posting: https://github.com/music-encoding/music-encoding/pull/848#issuecomment-944400479

If there is a final determination on how to represent a `3` mensuration in mensural music, then this will can also be used to display the mensuration sign in modern notation (as with other current mensuration signs).  This can be discussed within the Mensural SIG.  My proposal is to add these attributes in order to indicate the meaning of the `3`:

```
@tempus.num
@tempus.numbase
@prolatio.num
@prolatio.numbase
```

Since a `3` mensuration can mean either 3 semibreves in the time of 2 at the tempus rhythm level (most common for cut-C mensuration), or 3 minims in the time of two at the prolatio rhythmic level (rare but possible in cut-C, or the only option in O mensuration).

It still may be difficult to go from the attributes to a graphical symbol, however.    For the most part, if `@tempus.num="3"` and `@tempus.numbase="2"` are present or `@prolatio.num="3"` and `@prolatio.numbase="2"` are present, then a `3` should be displayed.

However @rettinghaus posted an example where the `3` mensuration is show as `cut-c-3`:

https://github.com/music-encoding/music-encoding/issues/785

<img width="103" alt="Screen Shot 2021-10-18 at 12 20 11 AM" src="https://user-images.githubusercontent.com/3487289/137685843-b2a2061b-ac91-4949-9982-6ca0bf4a43be.png">

Here is an earlier manuscript version:

<img width="310" alt="Screen Shot 2021-10-18 at 12 21 00 AM" src="https://user-images.githubusercontent.com/3487289/137686186-8f648d59-3141-495d-b751-c041c3d26b9b.png">

Notice that the "3" because "cut-c-3" in the printed version.  "cut-c-3" was probably used because "3" by itself has not meaning: there must be an existing mensuration sign which the sesquialtera is being applied to (typically O or cut-C in Josquin era).   The publisher (Petrucci?) was probably concerned about this ambiguity and printed it as "cut-c-3" for clarity.  However, this is a confusion, particularly in older music where such a sign (usually without the slash), has a very different meaning.

More background in issue https://github.com/music-encoding/music-encoding/issues/686

**Comment by craigsapp** [comment]
> This is why I keep suggesting embedded SVG as a possible solution. Using SVG conveys absolutely no semantic meaning, so you can put anything you like in it, and the encoding doesn't have to try and interpret an increasing number of (possibly conflicting) semantic meanings.

Resorting to an SVG graphic for mensuration signs is not appropriate as the primary/only method of representation mensuration signs in CMN, although for some editorial styles it may be suitable.  In any case, I will point out that it is irrelevant in the long run: It is not possible to display a `3` mensuration in mensural MEI yet.  Once that is possible, then the mensuration sign can also be displayed in CMN using verovio without a problem.

I cited several modern editions which preserve mensuration signs in CMN notation, in particular within the staff with a similar graphical behavior to time signatures.  It is more inappropriate to assign meter signatures to mensural music in CMN than to preserve the mensuration sign.   Meaning has not been fully removed from the mensuration signs in CMN, with the tempo meaning in particular still being relevant.  Compare this to common-time and cut-time in CMN, which are graphical symbols derived from mensural music with almost the same meanings as in the original mensural notation.  These are not represented as SVG images in MEI, so why should the other mensural signs be demoted to random graphical images?

Interpreting the absolute durations of mensural notes is not very ambiguous, but going further to interpret modern equivalents of tempo and meter from the mensuration signs can be ambiguous,  Going from cut-c to 3 means that the duration of every three whole notes in "3" have the same duration as two whole notes in cut-c.  Verovio currently does not understand this meaning, but it could be made to do so as a metric modulation in this particular case.  Can `<tempo>` represent a relative tempo change (speed up 1.5 times)?  As an aside, there is currently a problem displaying verbose attribute description on the element pages:
     https://music-encoding.org/guidelines/v4/elements/tempo.html

In the Tasso in Music project scores, the modern score display uses time signatures, with metric modulation marks above time signatures:

https://www.tassomusic.org/work/?id=Trm0602d

<img width="273" alt="Screen Shot 2021-10-18 at 12 41 11 AM" src="https://user-images.githubusercontent.com/3487289/137689169-c48a921a-e5cc-4d50-a111-0a205f35ff2e.png">

https://repository.royalholloway.ac.uk/items/6c44a5c0-7b2f-c907-0b76-ed699eecc929/1/K4c8_010.jpg?.vi=livNavTreeViewer&_sl.t=true

<img width="373" alt="Screen Shot 2021-10-18 at 12 51 59 AM" src="https://user-images.githubusercontent.com/3487289/137690250-da4b24b5-8655-46cd-b671-541e8abc4ade.png">

Note that the sign is `C3` is used instead of `3` for sesquialtera, similar to the Petrucci print, but this would be a confusing notation before 1500, since `C3`does not mean sesquialtera in earlier mensural music.  So seeing the mensuration signs of the original music may also be useful for placing the music historically.

How to transcribe the mensurations will be an editorial decision, from preserving them as in the case of the New Josquin Edition (CMN notation without mention/display of a modern time signature), to resolving to a modern time signature and showing a metric modulation as in the Tasso in Music project edition.   Another style that I have seen closer to what you propose is a hybrid one similar to the Fitzwilliam Virginal example, where there is a modern time signature, and the original mensuration sign is shown above it.   VerovioText has mensural signs (or I added them to my copy of verovio) for displaying mensuration signs in `<dir>` elements, which could be used for such cases (this is basically the convenient form of an SVG encoding).  But note that `<dir>` cannot be displayed `within` the staff in verovio (that I am aware of).

The editorial style will depend on the target audience:  singers not familiar with mensural music will prefer modern time signatures with explicit metric modulation signs, while singers familiar with mensural music will prefer to see the original mensuration signs without further interpretation.
